src/main/java/com/southernartsoap/repository/ProductRepository.java

```diff
@@ -2,13 +2,20 @@
 
 import java.util.List;
 
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
 import org.springframework.stereotype.Repository;
 
 import com.southernartsoap.model.Product;
 
 @Repository
 public interface ProductRepository extends CrudRepository<Product, Long> {
-    List<Product> findAll();
-    Product findById(long id);
+	List<Product> findAll();
+
+	Product findById(long id);
+
+	List<Product> findByCategory(String category);
+
+	@Query("SELECT DISTINCT p.category FROM Product p")
+	List<String> findDistinctCategories();
```

Comment:
Why does findDistinctCategories take no arguments?

Reply:
since it just looks through all the products and returns the unique categories